include/aws/s3/private/s3_util.h

```diff
@@ -144,6 +144,8 @@ AWS_S3_API
 extern const struct aws_byte_cursor g_range_header_name;
 
 extern const struct aws_byte_cursor g_if_match_header_name;
+AWS_S3_API
+extern const struct aws_byte_cursor g_request_id_header_name;
```

Comment:
do we really want to expose symbols for this header value?
if we need those headers in test should we just make them constant exposed in the header instead of externs?